src/com/goide/runconfig/testing/coverage/GoCoverageProgramRunner.java

```diff
@@ -32,13 +36,57 @@
 import com.intellij.execution.runners.GenericProgramRunner;
 import com.intellij.execution.runners.RunContentBuilder;
 import com.intellij.execution.ui.RunContentDescriptor;
+import com.intellij.notification.*;
 import com.intellij.openapi.fileEditor.FileDocumentManager;
+import com.intellij.openapi.vfs.VirtualFile;
 import com.intellij.util.ObjectUtils;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
+import javax.swing.event.HyperlinkEvent;
+
 public class GoCoverageProgramRunner extends GenericProgramRunner {
   private static final String ID = "GoCoverageProgramRunner";
+  private static final String GO_GET_BINARY_LINK = "goGetPackageCoverageLink";
+  private static final String GO_GET_BINARY_NAME = "github.com/haya14busa/goverage";
+  private static final String GO_GET_BINARY = "goverage";
+
+  private class CoveragePatcher extends GoTestRunningState.ExecutorPatcher {
+    private GoTestRunConfiguration myConfiguration;
+    private CoverageEnabledConfiguration myCoverageConfiguration;
+    private VirtualFile myPackageCoverageExecutable;
+
+    public CoveragePatcher(CoverageEnabledConfiguration coverageConfiguration, GoTestRunConfiguration configuration) {
```

Comment:
The `patcher` is supposed to patch executor, it should not know anything about TestRunConfiguration